### PR TITLE
2.4.2

### PR DESCRIPTION
**Pull request by 0xAsimetriq**
# Changes
- feat: add `@web3modal/sign-html` package
- feat: add `@web3modal/sign-react` package
- chore: move `from yarn@1` to `npm@latest`
- chore: dependency updates

### CODE REVIEW

File: package-lock.json
Comment (ignaciosantise) [inline package-lock.json:1]: @0xAsimetriq i think this file shouldn't be here, right? It was probably added after the change in `pre-publish` script

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "prettier": "prettier --check .",
     "typecheck": "npm run typecheck --ws --if-present",
     "new-version": "lerna version --no-git-tag-version --exact",
-    "pre-publish": "npm run lint; npm run typecheck; npm run prettier; npm run new-version; npm run build;",
+    "pre-publish": "npm run lint; npm run typecheck; npm run prettier; npm run build; npm run new-version; npm install",
```
Comment (ignaciosantise) [inline package.json:14]: `yarn install` instead of `npm install`?

File: package-lock.json
Comment (ignaciosantise) [reply package-lock.json:1]: ah you removed the yarn.lock. deleting this

Comment (0xAsimetriq) [reply package-lock.json:1]: Yeh, moved from yarn to npm (we were using yarn v1 so had to either upgrade that or move to npm or pnpm, but npm was easiest transition)

File: laboratory/src/pages/with-sign-api/react.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+export default function WithSignReactPage() {
+  const [modalOpen, setModalOpen] = useState(false)
+  const session = useSession()
+  const { disconnect } = useDisconnect({
+    topic: session?.topic as string,
+    reason: getSdkError('USER_DISCONNECTED')
+  })
+  const { connect, data } = useConnect({
+    requiredNamespaces: {
```
Comment (ignaciosantise) [inline laboratory/src/pages/with-sign-api/react.tsx:17]: you could move this to the constants file, it's also used in `WithSignHtmlPage`

File: laboratory/src/pages/with-sign-api/react.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+  }, [data?.topic])
+
+  return (
+    <>
+      <Card css={{ maxWidth: '400px', margin: '100px auto' }} variant="bordered">
+        <Card.Body css={{ justifyContent: 'center', alignItems: 'center', height: '200px' }}>
+          {session ? (
+            <>
+              {/* <Button shadow color="primary" onPress={onSignMessage}>
```
Comment (ignaciosantise) [inline laboratory/src/pages/with-sign-api/react.tsx:46]: do you need this commented code?

File: laboratory/src/pages/with-sign-api/react.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          <Text color="grey">{JSON.stringify(data, null, 2)}</Text>
+        </Modal.Body>
+      </Modal>
+
+      <Web3ModalSign
+        projectId={projectId}
+        modalOptions={{
+          themeMode: getTheme()
+        }}
+        metadata={{
```
Comment (ignaciosantise) [inline laboratory/src/pages/with-sign-api/react.tsx:76]: you can import this data from `data/Constants.ts`

File: rollup.config.js [hunk 5]
```diff
@@ -34,11 +29,26 @@ export default function createConfig(packageJson) {
...
     {
       input: './index.ts',
-      plugins: [replacePlugin, litCssPlugin, minifyHtml.default(), esbuildPlugin],
-      output: [{ file: './dist/index.js', format: 'es', ...output }]
+      plugins,
+      output: [{ file: './dist/index.es.js', format: 'es', ...output }]
     }
+    // {
+    //   input: './index.ts',
+    //   plugins: [...plugins, json(), polyfillNode(), commonjs(), resolve({ browser: true })],
```
Comment (ignaciosantise) [inline rollup.config.js:42]: need this code?

**Pull request merged**